### DISCUSSION TITLE
Don't return junction row when you don't have access to related m2a/m2m item

### DISCUSSION THREAD

**Discussion by ddcollins** [directus/directus#7364] [Feature Requests]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

I previously reported an issue that with an Item Permission rule of:
```
{
    "status": {
        "_eq": "published"
    }
}
```

M2A relation draft items were still being returned. This has since been partially resolved, but what I am seeing now is that the pivot table record is still being returned, which may or not be intentional. The problem in our case is we don't want that pivot table record to be returned because it will break the way we serve content to user.

See attached images:
![Image 2021-08-11 at 3 32 00 PM](https://user-images.githubusercontent.com/38921476/129099426-ca3085b5-c1bd-4879-bf71-3de37eb40e0e.jpg)
![Image 2021-08-11 at 3 32 52 PM](https://user-images.githubusercontent.com/38921476/129099440-cae54aba-419b-4b1e-803f-b6626bbcf37e.jpg)
![Image 2021-08-11 at 3 31 19 PM](https://user-images.githubusercontent.com/38921476/129099451-029df294-1c09-425e-90fb-374e9fb4b98e.jpg)

As you can see the second step "Draft Step" has `status` of "Draft" and the API does not return the item under the step record, but still returns that pivot step record.

Would it be possible to have that pivot record also be removed if the child item is "Draft"? There is no way to have a `status` column on that pivot record. 

This may be intentional, and I'm wondering if I could just set an Item Permission rule for Lesson Steps to filter out steps with item that is empty?
```
{
    "item": {
        "_nempty": null
    }
}
```

### To Reproduce

1. Create 3 collections.
2. For one of the collections, create a M2A field that references the two other collections.
3. Set up the Item Permission rule on the collections
```
{
    "status": {
        "_eq": "published"
    }
}
```
4. Query from the API on those fields.

### What version of Directus are you using?

9.0.0-rc.89

### What version of Node.js are you using?

directus/directus:9.0.0-rc.89

### What database are you using?

Postgres 11 Alpine

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying Directus?

Docker

**Comment by rijkvanzanten** [comment]
This is technically working as it should. You don't have permission to the `item` data, so `item` isn't returned. That being said, I can see how you'd want to filter out the junction record as well. In this case, I'm wondering if we could add a filter to the `steps` instead of `item` where `{ item: { status: { _neq: "draft" }}}` 🤔

**Reply by ddcollins** [reply]
Ah! Yes that sounds logical. I just added the filter rule but now am getting this error when I query:
![Image 2021-08-11 at 3 56 30 PM](https://user-images.githubusercontent.com/38921476/129102759-75469b33-7f2c-47f7-869d-97e93d9a92c6.jpg)

Which is probably because it's a M2A item and isn't guaranteed that `status` field exists on all the item types.

**Reply by ddcollins** [reply]
Also tried:
```
{
    "item": {
        "_neq": null
    }
}
```

but has no effect, as I believe it's looking at the database column value, versus the filtered `item` value from the other rule.

**Reply by rijkvanzanten** [reply]
> Which is probably because it's a M2A item and isn't guaranteed that status field exists on all the item types.


This is correct! I _guess_ you could do an `_or` with multiple `{ "item:lesson_slide": { "status": { _eq: "published" }}}` though that gets a little nuts too 🤔 The easiest solution would be to have a status field on the junction row itself

> but has no effect, as I believe it's looking at the database column value, versus the filtered item value from the other rule.

This is also correct. The value for `item` in the DB is technically not null, so this passes, even though you don't have the read access required to fetch the actual item

**Reply by ddcollins** [reply]
Hmm I think I'll go with the first solution of using `or`. I don't want to burden the content creators with another `status` field, especially since the steps junction collection grid isn't being displayed, so the only way to check the `status` value for them would be to click all the way through the relations.

**Reply by rijkvanzanten** [reply]
We already display custom fields on the junction table at the top of the edit drawer for many-to-many items. We should do the same thing for m2a items! That way, you could have the drawer for the related m2a start with

```
Status
[ Published         ]

-----------

.. m2a fields
```

(cc @benhaynes)

**Reply by benhaynes** [reply]
Yes, 100%!

**Reply by ddcollins** [reply]
@rijkvanzanten I do have 1 custom field `step_name` that gets shown in the drawer, but the custom fields go to the bottom of the drawer. If it went up to the top instead that would be ideal as it's more visible to the content creators. I could drop the `status` value off all the item types and just have the `status` on the step instead. Or hide the status field on the items and just default it to `publish` :shrug:


![Image 2021-08-11 at 4 47 29 PM](https://user-images.githubusercontent.com/38921476/129107855-cf445c3c-4ac3-4cee-b69e-cda81414ec90.jpg)

**Reply by rijkvanzanten** [reply]
Ahh perfect, so it already works then 😁 

In that case, I'd recommend just removing the status field from your individual item collections in favor of relying on this junction row!

**Reply by rijkvanzanten** [reply]
Would be interesting to look into an option that controls whether or not those junction rows show up at the top or bottom. I can see how it would be more useful to be at the top for this particular use case 🤔

**Reply by ddcollins** [reply]
Controllable would be great! I worry with our "Description" and "Content" fields getting extremely long the fields might be missed at the bottom of the drawer. Status is one we don't want them missing, especially since it has a defaulted value so a "required" flag wouldn't alert them that they never edited it appropriately.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
